File: src/transformers/features/misc/MiscCreatePruneTransformer.ts
import { MiscPruneProgrammer } from "../../../programmers/misc/MiscPruneProgrammer";

import type { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace MiscCreatePruneTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "misc.createPrune",
      write: MiscPruneProgrammer.write,
    });
}

File: packages/transform/src/features/misc/MiscCreatePruneTransformer.go
package misc

import (
	coremisc "github.com/samchon/typia/packages/core/src/programmers/misc"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var MiscCreatePruneTransformer = miscCreatePruneTransformerNamespace{}

type miscCreatePruneTransformerNamespace struct{}

func (miscCreatePruneTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coremisc.MiscPruneProgrammer.Write(call.TypeText)
}
